cmd/testvectors/main.go

```diff
@@ -0,0 +1,272 @@
+			tx     []byte
+		}{
+			{txtype: Spawn, tx: sdkWallet.Spawn(priv, templateWallet.TemplateAddress, spawnArgsWallet, nonce, opts...)},
+			{txtype: SelfSpawn, tx: sdkWallet.SelfSpawn(priv, nonce, opts...)},
+			{txtype: Spend, tx: sdkWallet.Spend(priv, destination, amount, nonce, opts...)},
+		}
+		for _, txPair := range txList {
+			logrus.Debugf("[%d] Generating test vector for %s %s %s", index, hrp, "wallet", txPair.txtype)
+			testVector := txToTestVector(txPair.tx, vm, index, amount, nonce, Wallet, txPair.txtype, destination.String(), hrp)
+			testVectors = append(testVectors, testVector)
+			index++
+		}
+
+		// MULTISIG
+		// include 1- and 2- of -1 and -2 (m of n)
+		for _, m := range []uint8{1, 2} {
+			for _, n := range []uint8{1, 2} {
+				// fill in the missing public keys (we already have one)
+				pubKeys := make([]core.PublicKey, n)
+
+				// frustratingly, we need the same list of pubkeys in a different format
+				// https://github.com/spacemeshos/go-spacemesh/issues/6061
+				edPubKeys := make([]ed25519.PublicKey, n)
+
+				pubKeys[0] = pubCore
+				edPubKeys[0] = ed25519.PublicKey(pub.Bytes())
+				if n > 1 {
+					pub2, _ := getKey()
+					pubKeys[1] = core.PublicKey(types.BytesToHash(pub2.Bytes()))
+					edPubKeys[1] = ed25519.PublicKey(pub2.Bytes())
+				}
+
+				spawnArgsMultisig := &templateMultisig.SpawnArguments{
+					Required:   m,
+					PublicKeys: pubKeys,
+				}
+
+				principal = core.ComputePrincipal(templateMultisig.TemplateAddress, spawnArgsMultisig)
+				vm.ApplyGenesis([]types.Account{{
+					Address: principal,
+					Balance: constants.OneSmesh,
+				}})
+
+				txList = []struct {
+					txtype typeTx
+					tx     []byte
+				}{
+					{txtype: Spawn, tx: sdkMultisig.Spawn(0, priv, principal, templateMultisig.TemplateAddress, spawnArgsMultisig, nonce, opts...).Raw()},
+					{txtype: SelfSpawn, tx: sdkMultisig.SelfSpawn(0, priv, templateMultisig.TemplateAddress, m, edPubKeys, nonce, opts...).Raw()},
+					{txtype: Spend, tx: sdkMultisig.Spend(0, priv, principal, destination, amount, nonce, opts...).Raw()},
+				}
+				for _, txPair := range txList {
+					logrus.Debugf("[%d] Generating test vector for %s %s %s %d of %d", index, hrp, "multisig", txPair.txtype, m, n)
+					testVector := txToTestVector(txPair.tx, vm, index, amount, nonce, Multisig, txPair.txtype, destination.String(), hrp)
+					testVectors = append(testVectors, testVector)
+					index++
+				}
+			}
+		}
+	}
+	return testVectors
+}
+
+func main() {
+	testVectors := generateTestVectors()
+
+	jsonData, err := json.MarshalIndent(testVectors, "", "  ")
+	if err != nil {
+		logrus.Fatalf("Error marshalling test vectors: %v", err)
+	}
+
+	fmt.Println(string(jsonData))
+}
+
+func getKey() (pub signing.PublicKey, priv signing.PrivateKey) {
+	// generate a random pubkey and discard the private key
+	edPub, edPriv, err := ed25519.GenerateKey(rand.New(rand.NewSource(rand.Int63())))
+	if err != nil {
+		log.Fatal("failed to generate ed25519 key")
```

Comment:
Avoid mixing different loggers: `logrus`, `log` and `fmt.Print` are all used in this app 🙂